addons/payment_authorize/models/payment_transaction.py

```diff
@@ -95,18 +95,37 @@ def _send_refund_request(self, amount_to_refund=None, create_refund_transaction=
                 create_refund_transaction=create_refund_transaction,
             )
 
-        refund_tx = super()._send_refund_request(
-            amount_to_refund=amount_to_refund, create_refund_transaction=False
-        )
+        # Only full refund is possible.
 
         authorize_API = AuthorizeAPI(self.acquirer_id)
-        rounded_amount = round(self.amount, self.currency_id.decimal_places)
-        res_content = authorize_API.refund(self.acquirer_reference, rounded_amount)
+        tx_details = authorize_API.get_transaction_details(self.acquirer_reference)
+
+        if tx_details and tx_details.get('err_code'):
+            res_content = {
+                'x_response_code': AuthorizeAPI.AUTH_ERROR_STATUS,
+                'x_response_reason_text': tx_details.get('err_msg')
+            }
```

Comment:
Does this always means it's a refund for a validation operation? Can't we just check the `self.operation` instead, if so?
You can add a comment to explain this.

Reply:
done :)